carbon-design-system/carbon
# fix(overflow-menu): improve overall a11y

Closes https://github.com/IBM/carbon-components/issues/666

Adds in recommended a11y additions to Overflow Menu component

#### Changelog

**New**

- `aria-expanded` toggle when it is open / close
- correct a11y roles and and aria-labels

**Changed**

- Event listener now listens for keydown, instead of keypress, so that `esc` can be used to close the overflow menu and return focus. The overflow menu can now also be opened with the space bar. Was not able to add in navigation via keyboard arrows, might want to open a separate issue for that.


#### Testing / Reviewing

- Navigate to Overflow Menu, use `enter` and `space` to open / close the menu. Use `esc` to close the menu
- Ensure `aria-expanded` toggles correctly
- @carmacleod, it would be awesome if you could test these changes with JAWS, as I do not have a machine that can test that.
